### PR TITLE
refactor: convert some block generators to goog.module

### PR DESCRIPTION
**Pull request by rachel-fenichel**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Part of #5598

### Proposed Changes

Converts to goog.module with named requires, and fixes missing requires, for 
- `generators/lua/colour.js`
- `generators/lua/lists.js`
- `generators/lua/logic.js`
- `generators/lua/loops.js`
- `generators/lua/math.js`
- `generators/lua/procedures.js`
- `generators/lua/text.js`
- `generators/lua/variables.js`
- `generators/lua/variables_dynamic.js`

### CODE REVIEW

File: generators/lua/loops.js [hunk 1]
```diff
@@ -34,75 +34,75 @@ Blockly.Lua.CONTINUE_STATEMENT = 'goto continue\n';
  * @return {string} Generated label or '' if unnecessary
  * @private
  */
-Blockly.Lua.addContinueLabel_ = function(branch) {
-  if (branch.indexOf(Blockly.Lua.CONTINUE_STATEMENT) !== -1) {
+Lua.addContinueLabel_ = function(branch) {
```
Comment (maribethb) [inline generators/lua/loops.js:37]: This should probably be converted to a local variable similar to getListIndex, I think?

**Review by maribethb — APPROVED** [review]

File: generators/lua/loops.js (hunk 1 above)
Comment (rachel-fenichel) [reply generators/lua/loops.js]: Good catch. Fixed.

**Comment by rachel-fenichel** [comment]
I fixed what you pointed out and also converted the three remaining files (text, variables, and variables_dynamic).

**Review by maribethb — APPROVED** [review]
new changes lgtm

**Pull request merged**